Error: jsonrpc2: code 10 message: invalid parameters

Environment:
Terraform v0.12.6
provider.xenorchestra v.0.2.0
XO: xo-server 5.51.1
Xen-Machine: XCP-ng 7.6.0

main.tf:
<pre>resource &quot;xenorchestra_vm&quot; &quot;foo&quot; {

  core_os = false
  cpu_cap = 2
  cpu_weight = 2
  cpus = 1
  memory_max = 1073733632
  name_label = &quot;someText&quot;
  name_description = &quot;someText&quot;
  cloud_config = &quot;${xenorchestra_cloud_config.devops.template}&quot;
  template = &quot;someNumericHashValue&quot;
  network {
    network_id = &quot;someNumericHashValue&quot;
  }

  disk {
    sr_id = &quot;someNumericHashValue&quot;
    name_label = &quot;someText&quot;
    size = 32212254720
  }
}
</pre>

terraform validate: Success! The configuration is valid.
terraform plan: Creates an execution plan and is ready to apply
terraform apply: Is fine until it asks for confirmation then it will try to create and fails on cmdline
<pre>
Error: jsonrpc2: code 10 message: invalid parameters

  on main.tf line 8, in resource "xenorchestra_vm" "foo":
   8: resource "xenorchestra_vm" "foo" {
</pre>

Even with a cloud config set to an empty string (like this cloud_config = "") terraform will run fine until the jsonrpc2 error is thrown.

[terraform.log](https://github.com/terra-farm/terraform-provider-xenorchestra/files/3835660/terraform.log)

So my guess is that the disk sr_id or network_id is incorrect and that is causing the API to return with invalid parameters. Although when I try to reproduce that in the acceptance tests I get a more obvious error.

```
ddelnano@ddelnano-desktop:~/code/terraform-provider-xenorchestra (master) $ make testacc
TF_ACC=1 go test ./... -v
?       github.com/ddelnano/terraform-provider-xenorchestra     [no test files]
=== RUN   TestGetPIFByDevice
--- PASS: TestGetPIFByDevice (0.60s)
=== RUN   TestGetTemplate
--- PASS: TestGetTemplate (1.20s)
=== RUN   TestThatUnmarshalingWorks
--- PASS: TestThatUnmarshalingWorks (0.00s)
=== RUN   TestUnmarshalingVmObject
--- PASS: TestUnmarshalingVmObject (0.00s)
PASS
ok      github.com/ddelnano/terraform-provider-xenorchestra/client      (cached)
=== RUN   TestAccXenorchestraDataSource_pif
--- PASS: TestAccXenorchestraDataSource_pif (2.34s)
=== RUN   TestAccXenorchestraDataSource_template
--- PASS: TestAccXenorchestraDataSource_template (2.58s)
=== RUN   TestAccXenorchestraCloudConfig_readAfterDelete
--- PASS: TestAccXenorchestraCloudConfig_readAfterDelete (1.21s)
=== RUN   TestAccXenorchestraCloudConfig_create
--- PASS: TestAccXenorchestraCloudConfig_create (0.93s)
=== RUN   TestAccXenorchestraCloudConfig_updateName
--- PASS: TestAccXenorchestraCloudConfig_updateName (1.44s)
=== RUN   TestAccXenorchestraCloudConfig_updateTemplate
--- PASS: TestAccXenorchestraCloudConfig_updateTemplate (1.61s)
=== RUN   TestAccXenorchestraCloudConfig_import
--- PASS: TestAccXenorchestraCloudConfig_import (1.09s)
=== RUN   TestAccXenorchestraVm_create
[DEBUG] VM params map[string]interface {}{"memoryMax":1073733632, "existingDisks":map[string]interface {}{"0":map[string]interface {}{"name_label":"Ubuntu Bionic Beaver 18.04_imavo", "size":32212254720, "$SR":"f469400-4a2b-5624-cf62-61e522e50ea1"}}, "VIFs":[]map[string]string{map[string]string{"network":"d225cf00-36f8-e6d6-6a29-02636d4de56b"}}, "bootAfterCreate":true, "name_label":"Name", "name_description":"description", "template":"2dd0373e-0ed5-7413-a57f-1958d03b698c", "cloudConfig":"template", "coreOs":false, "CPUs":1, "cpuCap":interface {}(nil), "cpuWeight":interface {}(nil)}--- FAIL: TestAccXenorchestraVm_create (3.11s)
    testing.go:569: Step 0 error: errors during apply:

            Error: jsonrpc2: code 1 message: no such object f469400-4a2b-5624-cf62-61e522e50ea1

                      on /tmp/tf-test315550669/main.tf line 16:
                                (source code not available)


FAIL
FAIL    github.com/ddelnano/terraform-provider-xenorchestra/xoa 14.346s
Makefile:22: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

So it would help if I could find a way to reproduce this so that we can improve the logging, add test cases, etc.

Can you look at the XOA api logs to see if that gives you anymore useful information or have you tried doing a packet capture to see what API request is returning the invalid params. That would be my next step if I had access to your XOA instance.

--- a/client/client.go
+++ b/client/client.go
@@ -102,7 +102,9 @@ func (c *Client) FindFromGetAllObjects(obj XoObject) (interface{}, error) {
 		panic(fmt.Sprintf("XO client does not support type: %v", t))
 	}
 	params := map[string]interface{}{
-		"type": xoApiType,
+		"filter": map[string]string{
+			"type": xoApiType,
+		},
 	}
 	var objsRes struct {
 		Objects map[string]interface{} `json:"-"`

--- a/client/vm.go
+++ b/client/vm.go
@@ -113,7 +113,9 @@ func (c *Client) DeleteVm(id string) error {
 
 func (c *Client) GetVm(id string) (*Vm, error) {
 	params := map[string]interface{}{
-		"type": "VM",
+		"filter": map[string]string{
+			"type": "VM",
+		},
 	}
 	var objsRes allObjectResponse
 	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)

--- a/modules/ROOT/pages/index.adoc
+++ b/modules/ROOT/pages/index.adoc
@@ -7,7 +7,7 @@ The provider needs to be configured with the proper credentials before it can be
 
 ** Terraform 0.12+
 ** Go 1.12 (to build the provider plugin)
-** Xen Orchestra 5.31.2+
+** Xen Orchestra 5.50.1
 
 == Using the provider
 

